hubstorage/jobq.py

```diff
@@ -31,6 +31,13 @@ def push(self, spider, **jobparams):
                 raise DuplicateJobError()
             raise
 
+    def jobsummary(self, jobkeys, jobmeta):
+        """Fetch selected job metadata fields for selected jobs."""
+        if not isinstance(jobkeys, (list, tuple)):
+            raise TypeError("jobkeys must be a list or a tuple")
+        return list(self.apiget(('jobsummary',),
```

Comment:
Other methods return a generator, because the output is potentially infinite. Limited by count, but still. Here you get at most the number of the keys you pass in, if I had to guess, I'd say that most of the time it would be only a few or even just one.

That was the reasoning behind my decision.